src/schema/resolvers/auth/mutation.resolver.js

```diff
@@ -0,0 +1,62 @@
+import { UserInputError, AuthenticationError } from "apollo-server";
+import { hash, createToken, isValid } from "../../../utils/auth";
+
+const signup = async (parent, args, context) => {
+  // validate that all usernames are unique
+  const { data: allUsers } = await context.database.users.getAll();
+
+  if (allUsers.find((user) => user.username === args.username)) {
+    throw new UserInputError("Username already exists.");
+  }
+
+  const password = await hash(args.password);
+  const payload = {
+    username: args.username,
+    password,
+    firstName: args.firstName,
+    lastName: args.lastName,
+    phoneNumber: "",
```

Comment:
These were optional right?
args.phoneNumber ?? "' etc

Reply:
Oh wait. Do we want them to sign up with phone number and email?